master/internal/rm/kubernetesrm/kubernetes_resource_manager.go

```diff
@@ -124,8 +125,17 @@ func (k ResourceManager) ResolveResourcePool(
 		return defaultComputePool, nil
 	}
 
+	resp, err := k.GetResourcePools(ctx, &apiv1.GetResourcePoolsRequest{})
+	if err != nil {
+		return "", err
+	}
+	rpConfig, err := rmutils.GetResourcePoolsResponseToConfig(resp)
```

Comment:
I think this would read better and be more flexible/usable if it was named something like `ResourcePoolsToConfig` and accepted `resp.ResourcePool` instead.